### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by olljanat** [burmilla/os#174] [General]
# About this release
This is first v2.x release version which replaces all v1.9.x versions.
Target is also have long lifetime on it so new versions will not be released unless there is critical bugs to be fixed, instead of new Docker versions will be made available by adding them to [here](https://github.com/burmilla/os-services/tree/v2.0.0).

Upgrade from v1.9.x versions/Rancher OS:
```bash
sudo ros os list --update
sudo ros os upgrade
```
And after reboot run also this command which upgrades console (any changes you have done to it **WILL be lost** so take backups first):
```bash
sudo ros console switch default
```
# Release notes compared to v1.9.x versions
* Kernel 5.10.211
* Buildroot 2023.02.10
* [Docker 24.0.9](https://github.com/moby/moby/releases/tag/v24.0.9) by default.
  * [Docker 25.0.3](https://github.com/moby/moby/releases/tag/v25.0.3) optionally available.
* Added apparmor support and enabled by default
* Removed experimental selinux support
* System-docker/engine bridge is disabled by default
* User Docker is not anymore included to ISO but is downloaded from internet on first boot.
  * If user has defined `rancher.docker.engine` in cloud-init, that version will be downloaded instead of default.
* VMware and QEMU tools upgraded to latest version.
* VirtualBox tools removed for now because building it did not work. Please, open issue if you need those.
* Added ipset to network and console system containers
* SSH security hardenings based [ssh-audit](https://github.com/jtesta/ssh-audit) report.
* NTP servers and hostname can be set with DHCP.

<hr /><em>This discussion was created from the release <a href='https://github.com/burmilla/os/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by PrplHaz4** [comment]
Great work on this @olljanat!  Thanks so much for keeping the project going.  My upgrade went very smoothly - I haven't observed anything strange so far.

Not sure if this is intended or not, but my user docker was not updated as part of the upgrade.  Updating user docker to 25.03 per the user guide seems to have worked great so far.

```
rancher@srv-rancher:~$ sudo ros -v
version v2.0.0 from os image burmilla/os:v2.0.0

rancher@srv-rancher:~$ system-docker --version
Docker version 17.06.107, build a4ecad9d

rancher@srv-rancher:~$ docker --version
Docker version 20.10.22, build 3a2c30b

rancher@srv-rancher:~$ sudo ros engine list --update
enabled  docker-24.0.9
disabled docker-25.0.3

sudo ros engine switch docker-25.0.3

rancher@srv-rancher:~$ sudo ros engine list --update
disabled docker-24.0.9
current  docker-25.0.3

rancher@srv-rancher:~$ docker --version
Docker version 25.0.3, build 4debf41
```

**Reply by olljanat** [reply]
Docker should update automatically to `24.0.9` on OS upgrade but it might be that if you had changed version with `ros engine switch` earlier then it comes as default because version is stored to cloud-init.
